packages/babel-plugin-transform-parameters/src/params.js

```diff
@@ -52,18 +51,22 @@ export default function convertFunctionParams(path, loose) {
   const { node, scope } = path;
 
   const state = {
-    iife: false,
-    scope: scope,
+    stop: false,
+    needsOuterBinding: false,
+    shadowedParams: new Set(),
```

Comment:
We can move `shadowedParams` out of `state` if we don't intend it accessed by `visitIdentifier`.